core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -985,29 +986,56 @@ public boolean decide(String workflowId) {
+                                 * Now find nested subworkflows that also need to have their tasks skipped
+                                 */
+                                for(Task workflowTask : workflow.getTasks()) {
+                                	if(TaskType.SUB_WORKFLOW.name().equals(workflowTask.getTaskType()) && StringUtils.isNotBlank(workflowTask.getSubWorkflowId())) {
+                                   		Workflow subWorkflow = executionDAOFacade.getWorkflowById(workflowTask.getSubWorkflowId(), true);
+                                		if(subWorkflow != null) {
+                                			skipTasksAffectedByTerminateTask(subWorkflow);
+                                		}
+                                	}
+                                }
+                                executionDAOFacade.updateTasks(terminateTasksToBeUpdated);
+                                if(workflowInstance.getStatus().equals(WorkflowStatus.COMPLETED)) {
+                                	completeWorkflow(workflow);
+                                } else {
+                                    workflow.setStatus(workflowInstance.getStatus());
+                                	terminate(workflow, new TerminateWorkflowException("Workflow is FAILED by TERMINATE task: " + task.getTaskId()));
+                                }
+                                return true;
                             }
-                            return true;
-                        }
-                        deciderService.externalizeTaskData(task);
-                        tasksToBeUpdated.add(task);
-                        stateChanged = true;
-                    } else if (SUB_WORKFLOW.name().equals(task.getTaskType()) && task.getStatus().equals(IN_PROGRESS)) {
-                        // Verifies and updates the task inplace, based on the Subworkflow and parent Workflow state,
-                        // and continues with the current decide.
-                        if (updateParentWorkflow(task, workflow)) {
+                            deciderService.externalizeTaskData(task);
                             tasksToBeUpdated.add(task);
                             stateChanged = true;
+                        } else if (SUB_WORKFLOW.name().equals(task.getTaskType()) && task.getStatus().equals(IN_PROGRESS)) {
+                            // Verifies and updates the task inplace, based on the Subworkflow and parent Workflow state,
+                            // and continues with the current decide.
+                            if (updateParentWorkflow(task, workflow)) {
+                                tasksToBeUpdated.add(task);
+                                stateChanged = true;
+                            }
                         }
+                    } catch (Exception e) {
+                        throw new ApplicationException(Code.INTERNAL_ERROR, String.format("Unable to start system task: %s", workflowSystemTask.getName()), e);
```

Comment:
This exception message: ```Unable to start system task:``` is not relevant, as we're not starting any system tasks here.